generate.ts

```diff
@@ -732,13 +732,40 @@ function templatizeTree(locals: { [key: string]: any }, sourcePath: string, dest
     for (var i = 0; i < filesInDir.length; i++) {
         var filename = filesInDir[i];
         var file = sourcePath + "/" + filename;
+
+        var folderExcluded = false;
+        if(excludeFolders != null)
+        {
+            for(var excludedFolderIndex = 0; excludedFolderIndex < excludeFolders.length; excludedFolderIndex++)
+            {
+                if(excludeFolders[excludedFolderIndex] == filename)
+                {
+                    folderExcluded = true;
+                    break;
+                }
+            }
+        }
+
+        var fileExcluded = false;
+        if(excludeFiles != null)
+        {
+            for(var excludedFileIndex = 0; excludedFileIndex < excludeFiles.length; excludedFileIndex++)
+            {
+                if(excludeFiles[excludedFileIndex] == filename)
+                {
+                    fileExcluded = true;
+                    break;
+                }
+            }
+        }
+
         if (fs.lstatSync(file).isDirectory()) {
-            if (excludeFolders != null && excludeFolders.includes(filename))
```

Comment:
Do you know why .includes() wasn't working? Some sort of difference in the execution environment?

Reply:
I think it's a mix of two issues: 1.) the build machines for certain private templates aren't recognizing the includes 2.) since javascript is interpreted, most of the other languages won't hit this issue since excludeFolders is null for most typical cases. 

So one alternative solution is tracking down why the includes javascript function does not work as expected on our build agents (I just updated Node JS and that didn't seem to do the trick).